XMonad/Actions/WorkspaceNames.hs

```diff
@@ -185,3 +196,21 @@ workspaceNamePrompt conf job = do
                                 Just i -> i
         contains completions input =
           return $ filter (Data.List.isInfixOf input) completions
+
+setTag :: (WorkspaceId -> WorkspaceId) -> WindowSpace -> WindowSpace
+setTag remap ws = ws { W.tag = remap $ W.tag ws }
+
+ewmhWorkspaceNamesLogHook' :: ((WorkspaceId -> Maybe String) -> WorkspaceId -> String)
+                           -> X ()
+ewmhWorkspaceNamesLogHook' nameWorkspace =
+  map . setTag . nameWorkspace <$> getWorkspaceNames' >>=
+      ewmhDesktopsLogHookCustom
+
+ewmhWorkspaceNamesLogHook = ewmhWorkspaceNamesLogHook' getWorkspaceNameFromTag
+
+ewmhWorkspaceNames :: XConfig a -> XConfig a
+ewmhWorkspaceNames c = c { startupHook     = startupHook c +++ ewmhDesktopsStartup
+                         , handleEventHook = handleEventHook c +++ ewmhDesktopsEventHook
+                         , logHook         = logHook c +++ ewmhWorkspaceNamesLogHook
+                         }
+  where x +++ y = mappend y x
```

Comment:
I don't understand why there is EWMH code in `WorkspaceNames.hs`.  Anything with `ewmh` in the name should be moved out of this file and into `EwmhDesktops.hs`.

Also, your `ewmhWorkspaceNames` function is almost an exact copy of the `ewmh` in `EwmhDesktops.hs`.

Please fix these issues then update the `CHANGES.md` file.

Reply:
I think it'd be better to turn the `WindowSet -> WindowSet` in `EwmhDesktops` into `X (WindowSet -> WindowSet)` so that `WorkspaceNames` can just provide such a function and users would then be able to plug that into their `ewmhWhateverHookCustom`. This should simplify the code significantly.

See also #399. I'll be happy to make that change there. That is, if @IvanMalison still wants to pursue this pull request, or if there are other users who'd like to see it merged. I'm not using any pager that shows desktop names, personally, but I'd be very happy to see this supported in `WorkspaceNames` and `EwmhDesktops`.

Also, I'd like to remark that I'm mentioned as a maintainer of `WorkspaceNames` and that I would have reacted a few years earlier had I been pinged by either one of you. :-))

Reply:
I drafted this at https://github.com/liskin/xmonad-contrib/pull/1. Got to admit it's significantly simpler :-)

Reply:
Oh, I just realized that you already had the simple implementation in the cover letter of #105, but it wasn't possible back then. And two years later in #238 you fixed `ewmhDesktopsLogHookCustom` so that the simple implementation is possible, and this PR (#122) was never updated accordingly. Anyway, now that the simple implementation is possible, I'll just include it in #399.

Reply:
And a few months later I realized it's not as simple as I thought it was. With two separate hooks, it is, as one can pass the renaming function to the log hook only, but in #399 I deprecated the individual hooks and now I need two different remaps as in #105 again. 🤦
https://github.com/xmonad/xmonad-contrib/pull/399/commits/2e982f128943a776a2b857d85746a963520d9313
Not really a big deal I guess, but I do feel like an idiot. :-)